enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,608 @@
+          subnets:
+            - subnetA
+            - subnetB
+            - subnetC
+status:
+  availableReplicas: 2
+  conditions:
+  - message: Deployment has minimum availability.
+    reason: MinimumReplicasAvailable
+    status: "True"
+    type: Available
+  - message: "The IngressController subnets were changed from [\"\"] to [\"subnetA\", \"subnetB\", \"subnetC\"]. To effectuate this change, you must delete the service: `oc -n openshift-ingress delete svc/custom-default`; the service load-balancer will then be deprovisioned and a new one created. This will most likely cause the new load-balancer to have a different host name and IP address and cause disruption. To return to the previous state, you can revert the change to the IngressController: `oc -n openshift-ingress-operator patch ingresscontrollers/default --type=merge --patch='{\"spec\":{\"endpointPublishingStrategy\":{\"type\":\"LoadBalancerService\",\"loadBalancer\":{\"providerParameters\":{\"type\":\AWS\",\"aws\":{\"type\":\"Classic\",\"subnets\":[]}}}}}}'"
+    reason: OperandsProgressing
+    status: "True"
+    type: LoadBalancerProgressing
+```
+
+We have intentionally designed this feature to require manual deletion of the
+service for the following reasons:
+
+##### Reason #1: To mitigate risks associated with cluster admins providing an invalid annotation value.
+
+While the AWS Cloud Controller Manager (CCM) handles invalid annotations by
+producing errors and events, it is difficult for the Ingress Operator to
+decipher these CCM events _after_ the load balancer has been provisioned.
+Therefore, the Ingress Operator can't indicate to a cluster admin that an
+IngressController's load balancer is in a malfunctioning state (i.e., the
+service is not getting reconciled) while the service is already provisioned.
+
+When the IngressController is first created and the LoadBalancer-type
+service is not yet provisioned, these same invalid `Subnets` values will
+prevent the load balancer from being provisioned. The existing Ingress
+Operator logic will clearly indicate to the cluster admin that the load balancer
+failed to provision via the `LoadBalancerReady` status condition. In addition,
+the `LoadBalancerReady` condition will include the CCM event logs that indicate
+to the cluster admin that the `Subnets` value is invalid (see [Support Procedures](#support-procedures)
+for examples). Therefore, requiring manual service deletion provides a way
+for the Ingress Operator to produce a clear signal to the cluster admin
+that `Subnets` is invalid.
+
+See [Invalid Subnet Annotation Values](#invalid-subnet-annotation-values) for
+examples of invalid subnet annotation values.
+
+##### Reason #2: To mitigate upgrade compatibility issues.
+
+**Note**: Directly configuring the `service.beta.kubernetes.io/aws-load-balancer-subnets`
+on the LoadBalancer-type service is not supported and has never
+been. However, this enhancement is designed to prevent cluster disruption upon
```

Comment:
```suggestion
on the LoadBalancer-type service is not supported and never has 
been. However, this enhancement is designed to prevent cluster disruption upon
```